packages/compass-preferences-model/src/storage.spec.ts

```diff
@@ -136,6 +136,44 @@ describe('storage', function () {
       ).to.deep.equal(getDefaultPreferences());
     });
 
+    it('when invalid json is stored, it sets the defaults', async function () {
+      const storage = new StoragePreferences(tmpDir);
+
+      const preferencesFile = getPreferencesFile(tmpDir);
+      await fs.mkdir(getPreferencesFolder(tmpDir));
+      await fs.writeFile(preferencesFile, '{}}', 'utf-8');
+
+      // Ensure it exists
+      expect(async () => await fs.access(preferencesFile)).to.not.throw;
+
+      await storage.setup();
+
+      expect(
+        JSON.parse((await fs.readFile(preferencesFile)).toString())
+      ).to.deep.equal(getDefaultPreferences());
+    });
+
+    it('when invalid value is stored, it sets the defaults', async function () {
+      const storage = new StoragePreferences(tmpDir);
+
+      const preferencesFile = getPreferencesFile(tmpDir);
+      await fs.mkdir(getPreferencesFolder(tmpDir));
+      await fs.writeFile(
+        preferencesFile,
+        JSON.stringify({ enableMaps: 'a string' }),
+        'utf-8'
+      );
+
+      // Ensure it exists
+      expect(async () => await fs.access(preferencesFile)).to.not.throw;
+
+      await storage.setup();
+
+      expect(
+        JSON.parse((await fs.readFile(preferencesFile)).toString())
+      ).to.deep.equal(getDefaultPreferences());
```

Comment:
this is definitely a bigger change and really nice to have. we can handle it separately as a follow up.

Reply:
Hm, that's also unexpected to me, I was expecting what Le Roux described as a way to handle that, not overriding all the preferences if one of them was not validated correctly

Reply:
If it is really hard to do and requires lots of code surgery I would say that this is already a step in the right direction, but ideally I think we wouldn't lose the other preferences if it can be helped.